pkg/resource/logging-credentials/logging_operator_secrets.go

```diff
@@ -24,22 +24,23 @@ func LoggingCredentialsSecretMeta(lc loggedcluster.Interface) metav1.ObjectMeta
 }
 
 // Generate a random 20-characters password
-func genPassword() string {
+func genPassword() (string, error) {
 	const length = 20
 
-	rand.Seed(time.Now().UnixNano())
 	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
 		"abcdefghijklmnopqrstuvwxyz" +
 		"0123456789" +
 		"~=+%^*/()[]{}/!@#$?|")
 
-	var b strings.Builder
+	pass := make([]rune, length)
 	for i := 0; i < length; i++ {
-		b.WriteRune(chars[rand.Intn(len(chars))])
-
+		num, err := rand.Int(rand.Reader, big.NewInt(int64(len(chars))))
+		if err != nil {
+			return "", err
```

Comment:
We usually wrap errors for easier debugging when things go wrong
```suggestion
			return "", errors.WithStack(err)
```